### PR TITLE
MRG Min duration for stim channel synthesis = 1

### LINKED ISSUE #5249: Brainvision stim channel creation fails if event durations are 0
Stim channels for brainvision and EEGLAB files are created like this:

```
    for onset, duration, trigger in events:
        stim_channel[onset:onset + duration] = trigger
```

If the duration is 0, which it often is, this will result in no events being recorded.

I am not sure how to deal with this.

The events are still available to the user at `raw._events`, but `mne.find_events` etc won't work.

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
closes #5249

### CODE REVIEW

File: mne/io/utils.py [hunk 1]
```diff
@@ -241,6 +242,8 @@ def _synthesize_stim_channel(events, n_samples):
     """
     # select events overlapping buffer
     onset = events[:, 0]
+    events = events.copy()
+    events[events[:, 1] < 1, 1] = 1
```
Comment (agramfort) [inline mne/io/utils.py]: why are you changing second column?

File: mne/io/utils.py [hunk 2]
```diff
@@ -231,6 +231,7 @@ def _synthesize_stim_channel(events, n_samples):
     events : array, shape (n_events, 3)
         Each row representing an event as (onset, duration, trigger) sequence
         (the format returned by `_read_vmrk_events` or `_read_eeglab_events`).
+        Durations == 0 are converted to 1.
```
Comment (agramfort) [inline mne/io/utils.py]: Durations equal to 0 are converted to 1.

File: mne/io/utils.py (hunk 1 above)
Comment (jona-sassenhagen) [reply mne/io/utils.py]: ... cause it’s the duration column?

**Comment by agramfort** [comment]
second column is not duration its previous value when trigger channel
changed.

See:

https://github.com/mne-tools/mne-python/blob/master/mne/event.py#L630

**Comment by jona-sassenhagen** [comment]
Ah yes, sure. 
I am memory impaired. 

Well as you see, here, we take the 2nd column to be the duration for the purpose of extracting BV and EEGLAB data. See the docstring of the function we are in :)

It’s how these readers are written, considering the data sources.

**Comment by agramfort** [comment]
this is not how it should be done.

see my (unfinished efforts) in https://github.com/mne-tools/mne-python/pull/4936

basically the idea is get Annotations from the edf files (no more stim channel)
and have a function make them events by looking at onsets.

I think I already wrote this somewhere...

**Comment by jona-sassenhagen** [comment]
>no more stim channel)

I like that - but as you see in https://github.com/mne-tools/mne-python/issues/5249, both me and @mmagnuski had problems at workshops recently where BV people couldn't read in their files, and this is a very small fix (in fact suggested by Eric) ... So maybe we can still do it here (for BV and EEGLAB), so people can read in their files until we move over to Annotations.

**Comment by cbrnr** [comment]
I agree with @agramfort that it is a bit weird that our `events` array gets misused like that. But since this has been there for longer, fixing this hack to let people open files is better than not fixing it. However, this should be documented, and more importantly, the reader should be fixed to correctly fill the events array.

**Comment by agramfort** [comment]
@jona-sassenhagen can you put some XXX in the code so we keep a trace in the code of things that should be fixed? Something liek
```
# XXX : Hack to put duration in 2nd column. Should be fixed using Annotations. See discussion in #5274
```

**Comment by jona-sassenhagen** [comment]
As the docstring says, the duration already is in the 2nd column when it’s passed to this function. The problem fixed here is: BV files specify the duration in the marker file. But often, the duration will be zero. The corresponding events are dropped, which is undesirable (people get completely empty events arrays when they use mne find_events). 

This fix is simply for when reconstructing the stim channel, which already, for the BV reader on master, rests on the middle column being durations, all durations are set to min 1. 

So I’ll see where this happens is upstream, and can put the fixme there.

**Comment by jona-sassenhagen** [comment]
Fixme added, ready for merge :)

File: mne/io/utils.py (hunk 2 above)
Comment (larsoner) [reply mne/io/utils.py]: not addressed yet

**Comment by agramfort** [comment]
thx @jona-sassenhagen

**Pull request merged**